lib/solveEquation/stepThrough.js

```diff
@@ -50,9 +51,23 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   const MAX_STEP_COUNT = 20;
   let iters = 0;
 
+  // Checks if there are roots in the original equation before we
```

Comment:
I do have to make a second call because if the equation isn't factored yet (e.g. `x^2 + 3x + 2 =0`), we have to first factor and then look for roots in the `do ... while` loop. If the original equation is already factored (e.g. `(x+2) (x-2) = 0`), we don't have to go through any of the simplification steps and find the roots immediately.